resume-builder/src/TestComponent.js

```diff
@@ -0,0 +1,196 @@
+      "fluency": "Native speaker"
+    }],
+    "interests": [{
+      "name": "Wildlife",
+      "keywords": [
+        "Ferrets",
+        "Unicorns"
+      ]
+    }],
+    "references": [{
+      "name": "Jane Doe",
+      "reference": "Reference..."
+    }]
+  }
+  // TODO
+  // 1. Use immutability-helper package to handle nested data
+  // 2. Programatically create an input for each key
+  // 3. Setup append for array outputs && create
+  //    a button to clear all array contents
+  // 4. Add the JSON Resume field contents example
+  //    near the input element
+
+  // programatically create input elements
+  // componentDidMount() {
+  //   for (let i=0; i < Object.keys(this.state.basics).length; i++) {
+  //     // make a new input element for the object key
+  //     let newInput = document.createElement("input")
+
+  //     // let the user know what the box will fill in
+  //     let elementLabel = document.createElement("p")
+  //     elementLabel.innerText = Object.keys(this.state.basics)[i]
+
+  //     // just in case I need to access it
+  //     newInput.className = Object.keys(this.state.basics)[i]
+
+  //     // add the label and input to DOM for Basics section
+  //     document.querySelector('.basics').appendChild(elementLabel)
+  //     document.querySelector('.basics').appendChild(newInput)
+  //   }
+  // }
+
+  // this is bad, wouldn't use in final implementation
+  handleNameChange = e => {
+    e.preventDefault()
+    this.setState({
+      basics: {
+        name: e.target.elements[0].value,
+        label: e.target.elements[1].value,
+        picture: e.target.elements[2].value,
+        email: e.target.elements[2].value,
+        phone: e.target.elements[2].value,
+        website: e.target.elements[2].value,
+        summary: e.target.elements[2].value,
+        location: {
+          address: e.target.elements[2].value,
+          postalCode: e.target.elements[2].value,
+          city: e.target.elements[2].value,
+          countryCode: e.target.elements[2].value,
+          region: e.target.elements[2].value
+        }
+      }
+      // insert rest of resume elements here
+    })
+  }
+
+  render() {
+    return (
+      <div>
+        <div>
+          <form onSubmit={this.handleNameChange}>
+            <h3 className="basics">Basics</h3>
+            <p>Name</p>
+            <input type="textarea"/>
+            <p>Label</p>
+            <input></input>
+            <p>Picture</p>
+            <input></input>
+            <p>Email</p>
+            <input></input>
+            <p>Phone</p>
+            <input></input>
+            <p>Website</p>
+            <input></input>
+            <p>Summary</p>
+            <input></input>
+            <h3>Work</h3>
+
+            <h3>Volunteer</h3>
+
+            <h3>Education</h3>
+
+            <h3>Awards</h3>
+
+            <h3>Publications</h3>
+
+            <h3>Language</h3>
+
+            <h3>Interests</h3>
+
+            <h3>References</h3>
```

Comment:
The use the <h3> and <input> tags is repetitive here. I would reimplement using a loop that reads the headings from state to make this code less verbose.